packages/react/testing-library/src/pure.jsx

```diff
@@ -0,0 +1,128 @@
+// Copyright 2024 The Lynx Authors. All rights reserved.
+// Licensed under the Apache License Version 2.0 that can be found in the
+// LICENSE file in the root directory of this source tree.
+import { configure as configureDTL, getQueriesForElement, queries } from '@testing-library/dom';
+import { cloneElement, createRef, h, render as preactRender } from 'preact';
+import { useEffect } from 'preact/hooks';
+import { act } from 'preact/test-utils';
+
+import { __root } from '@lynx-js/react/internal';
+
+import { flushDelayedLifecycleEvents } from '../../runtime/lib/lynx/tt.js';
+import { clearPage } from '../../runtime/lib/snapshot.js';
+
+export function waitSchedule() {
+  return new Promise(resolve => {
+    requestAnimationFrame(() => {
+      setTimeout(resolve);
+    });
+  });
+}
+
+configureDTL({
+  asyncWrapper: async cb => {
+    let result;
+    await act(() => {
+      result = cb();
+    });
+    return result;
+  },
+  eventWrapper: cb => {
+    let result;
+    act(() => {
+      result = cb();
+    });
+    return result;
+  },
+});
+
+export function render(
+  ui,
+  {
+    queries,
+    wrapper: WrapperComponent,
+    enableMainThread = false,
+    enableBackgroundThread = true,
+  } = {},
+) {
+  if (!enableMainThread && !enableBackgroundThread) {
+    throw new Error(
+      'You must enable at least one thread for rendering (enableMainThread or enableBackgroundThread)',
+    );
+  }
+  const wrapUiIfNeeded = (innerElement) => (WrapperComponent
+    ? h(WrapperComponent, null, innerElement)
+    : innerElement);
+
+  const comp = wrapUiIfNeeded(ui);
+  const compMainThread = cloneElement(comp);
+  const compBackgroundThread = cloneElement(comp);
+
+  globalThis.lynxEnv.switchToMainThread();
+  __root.__jsx = enableMainThread ? compMainThread : null;
+  renderPage();
+  if (enableBackgroundThread) {
+    globalThis.lynxEnv.switchToBackgroundThread();
+    act(() => {
+      preactRender(compBackgroundThread, __root);
+      flushDelayedLifecycleEvents();
+    });
+  }
+
+  return {
+    container: lynxEnv.mainThread.elementTree.root,
+    unmount: cleanup,
+    rerender: (rerenderUi) => {
+      lynxEnv.resetLynxEnv();
+      return render(wrapUiIfNeeded(rerenderUi), {});
```

Comment:
should the `rerender` inherits the options of its `render`?
```suggestion
      return render(wrapUiIfNeeded(rerenderUi), {
        {
          queries,
          wrapper: WrapperComponent,
          enableMainThread,
          enableBackgroundThread,
        }
      });
```